src/Microsoft.ML.AutoML/AutoMlUtils.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.ML.AutoML
+{
+    internal static class AutoMlUtils
+    {
+        public static readonly ThreadLocal<Random> Random = new ThreadLocal<Random>(() => new Random());
+
+        public static void Assert(bool boolVal, string message = null)
```

Comment:
A few problems with this method:

1. Typically a method named `Assert` should be removed on `Release` builds. Normal .NET devs expect asserts to be debug only.
2. ML.NET already has a mechanism for checking bool values, and either throwing or asserting. Since this code is a part of ML.NET, it should be using those mechanisms.

Reply:
Nuked this and there's another copy we had that's used here and there... what a mess, thanks for pointing it out.